server/repository/src/db_diesel/assets/asset.rs

```diff
@@ -0,0 +1,271 @@
+use super::asset_row::{
+    asset::{self, dsl as asset_dsl},
+    AssetRow,
+};
+
+use diesel::{dsl::IntoBoxed, prelude::*};
+
+use crate::{
+    diesel_macros::{
+        apply_date_filter, apply_equal_filter, apply_sort, apply_sort_no_case, apply_string_filter,
+    },
+    repository_error::RepositoryError,
+    DBType, DateFilter, EqualFilter, Pagination, Sort, StorageConnection, StringFilter,
+};
+
+type Asset = AssetRow;
+
+pub enum AssetSortField {
+    SerialNumber,
+    InstallationDate,
+    ReplacementDate,
+    ModifiedDatetime,
+}
+
+pub type AssetSort = Sort<AssetSortField>;
+
+#[derive(Clone)]
+pub struct AssetFilter {
+    pub id: Option<EqualFilter<String>>,
+    pub serial_number: Option<StringFilter>,
+    pub category_id: Option<EqualFilter<String>>,
```

Comment:
would it need to have a `category` and `type` different to that on the `catalogue_item` though?
I'm thinking probably not - that the class, category and type are defined by the catalogue_item and are immutable

Reply:
I agree, was just thinking if we record it here it would make the filters easy.
I'll take them out and update the filters...